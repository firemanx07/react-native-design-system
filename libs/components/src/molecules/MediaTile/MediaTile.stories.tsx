import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import {
  StorybookRow,
  StorybookScreen,
  StorybookSection,
  StorybookSectionText,
} from '../../storybook';
import {
  MediaTile as MediaTileBase,
  PropsType as MediaTileProps,
} from './MediaTile';

const Story: ComponentMeta<typeof MediaTileBase> = {
  component: MediaTileBase,
  title: 'Molecules/MediaTile',
};
export default Story;

const commonProps = {
  source: { uri: 'https://picsum.photos/124/124' },
  size: 124,
  labels: {
    new: 'New',
    unlock: 'Freishalten',
  },
};

const Template: ComponentStory<typeof MediaTileBase> = (
  args: MediaTileProps
) => (
  <StorybookScreen title="MediaTile:">
    <StorybookSection>
      <MediaTileBase {...args} />
    </StorybookSection>
    <StorybookSectionText>Sample variants:</StorybookSectionText>
    <StorybookRow>
      <StorybookSection title="New - Comment">
        <MediaTileBase {...commonProps} isNew hasComment />
      </StorybookSection>
      <StorybookSection title="New - Selectable">
        <MediaTileBase {...commonProps} isNew isSelectable />
      </StorybookSection>
      <StorybookSection title="New - Selected">
        <MediaTileBase {...commonProps} isSelectable isNew isSelected />
      </StorybookSection>
    </StorybookRow>
    <StorybookRow>
      <StorybookSection title="Video">
        <MediaTileBase {...commonProps} videoDuration="01:22" />
      </StorybookSection>
      <StorybookSection title="Video - New - Comment">
        <MediaTileBase
          {...commonProps}
          videoDuration="01:22"
          isNew
          hasComment
        />
      </StorybookSection>
      <StorybookSection title="Unlock button">
        <MediaTileBase {...commonProps} showUnlockButton />
      </StorybookSection>
    </StorybookRow>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: [
      'uri',
      'size',
      'labels',
      'isNew',
      'hasComment',
      'isSelectable',
      'isSelected',
      'showUnlockButton',
      'videoDuration',
    ],
  },
};

const defaultArgs: MediaTileProps = {
  ...commonProps,
  isNew: false,
  hasComment: false,
  isSelectable: false,
  isSelected: false,
  showUnlockButton: false,
  videoDuration: '',
};

export const MediaTile = Template.bind({});
MediaTile.parameters = parameters;
MediaTile.args = {
  ...defaultArgs,
};
