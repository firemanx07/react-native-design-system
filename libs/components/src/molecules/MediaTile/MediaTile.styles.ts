import { styled, css } from '@proxym/themes';
import { rgba } from 'polished';

import {
  Badge,
  BadgeSize,
  BadgeVariant,
  Checkbox,
  MenuIcon,
  BaseButton,
  ButtonSize,
  ButtonVariant,
  LockIcon,
  BaseText,
  TextVariant,
  TextWeight,
} from '../../atoms';

enum Layer {
  One = 1,
  Two = 2,
  Three = 3,
}

type StyledCommonType = {
  size: number;
};

export const Container = styled.View<StyledCommonType>`
  height: ${({ size }) => size}px;
  width: ${({ size }) => size}px;
  overflow: hidden;
`;

export const MediaTileImage = styled.Image<StyledCommonType>`
  height: ${({ size }) => size}px;
  width: ${({ size }) => size}px;
`;

export const NewBadge = styled(Badge).attrs({
  size: BadgeSize.Small,
  variant: BadgeVariant.Square,
  textStyle: {
    letterSpacing: 1.5,
  },
})`
  background-color: ${({ theme }) => theme.ds.colors.success};
  position: absolute;
  top: 6px;
  right: 6px;
  z-index: ${Layer.One};
  padding-horizontal: 6px;
`;

export const StyledCommentIcon = styled(MenuIcon).attrs(({ theme }) => ({
  width: theme.ds.iconSize.primary,
  height: theme.ds.iconSize.primary,
  fill: theme.ds.colors.light,
}))`
  position: absolute;
  bottom: 4px;
  right: 4px;
  z-index: ${Layer.One};
`;

export const absoluteFillObject = css`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`;

export const Overlay = styled.View`
  ${absoluteFillObject};
  z-index: ${Layer.Two};
  background-color: ${({ theme }) => rgba(theme.ds.colors.dark, 0.5)};
`;

export const StyledCheckbox = styled(Checkbox)`
  position: absolute;
  bottom: 6px;
  right: 6px;
  z-index: ${Layer.Three};
`;

export const ButtonWrapper = styled.View`
  ${absoluteFillObject};
  justify-content: center;
  align-items: center;
  z-index: ${Layer.Three};
`;

export const ButtonUnlock = styled(BaseButton).attrs(({ theme }) => ({
  textStyle: {
    color: theme.ds.colors.light,
  },
  size: ButtonSize.Small,
  variant: ButtonVariant.Outline,
}))`
  height: 28px;
  border-color: ${({ theme }) => theme.ds.colors.light};
  background-color: transparent;
  align-self: center;
  padding-horizontal: ${({ theme }) => theme.ds.spacing.small}px;
`;

export const StyledLockKeyIcon = styled(LockIcon).attrs(({ theme }) => ({
  fill: theme.ds.colors.light,
}))``;

export const VideoDurationText = styled(BaseText).attrs({
  variant: TextVariant.S,
  weight: TextWeight.Bold,
})`
  color: ${({ theme }) => theme.ds.colors.light};
  text-shadow: ${({ theme }) => '0 0 2px ' + rgba(theme.ds.colors.dark, 0.5)};
  position: absolute;
  top: 5px;
  left: 8px;
  z-index: ${Layer.Two};
`;
