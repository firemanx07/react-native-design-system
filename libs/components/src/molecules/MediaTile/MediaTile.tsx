import React, { memo } from 'react';
import { ImageSourcePropType, ViewStyle, StyleProp } from 'react-native';

import { TestIDType } from '../../types';
import {
  ButtonWrapper,
  ButtonUnlock,
  StyledLockKeyIcon,
  Container,
  NewBadge,
  StyledCommentIcon,
  Overlay,
  StyledCheckbox,
  MediaTileImage,
  VideoDurationText,
} from './MediaTile.styles';

export type PropsType = {
  source: ImageSourcePropType;
  labels: {
    new: string;
    unlock: string;
  };
  isNew?: boolean;
  hasComment?: boolean;
  isSelectable?: boolean;
  isSelected?: boolean;
  showUnlockButton?: boolean;
  videoDuration?: string;
  size?: number;
  style?: StyleProp<ViewStyle>;
} & TestIDType;

export const MediaTile = ({
  source,
  size = 124,
  style,
  labels,
  isNew,
  hasComment,
  isSelectable,
  isSelected,
  showUnlockButton,
  videoDuration,
  testID,
}: PropsType) => {
  const renderUnlockButton = () => {
    if (!showUnlockButton) {
      return null;
    }
    return (
      <ButtonWrapper>
        <ButtonUnlock
          renderIcon={(iconSize: number) => (
            <StyledLockKeyIcon width={iconSize} height={iconSize} />
          )}
          disabled
        >
          {labels.unlock}
        </ButtonUnlock>
      </ButtonWrapper>
    );
  };

  return (
    <Container size={size} style={style} testID={testID}>
      {isNew && <NewBadge>{labels.new.toUpperCase()}</NewBadge>}
      {hasComment && !isSelectable && <StyledCommentIcon />}
      {(isSelected || showUnlockButton) && <Overlay />}
      {isSelectable && <StyledCheckbox isChecked={isSelected} />}
      {!!videoDuration && (
        <VideoDurationText>{videoDuration}</VideoDurationText>
      )}
      {renderUnlockButton()}
      <MediaTileImage size={size} source={source} />
    </Container>
  );
};

export default memo(MediaTile);
