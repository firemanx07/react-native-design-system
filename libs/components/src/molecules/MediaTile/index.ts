export { default as MediaTile } from './MediaTile';
