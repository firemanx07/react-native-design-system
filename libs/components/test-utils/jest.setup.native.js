import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();

console.warn = () => {};

Animated.timing = () => ({
  start: () => jest.fn(),
});
