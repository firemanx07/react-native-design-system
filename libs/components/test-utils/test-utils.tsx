import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
