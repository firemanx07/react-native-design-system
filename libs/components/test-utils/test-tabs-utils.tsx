import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
