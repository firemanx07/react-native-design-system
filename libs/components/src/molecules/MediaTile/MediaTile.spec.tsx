import { testAllStories } from '../../../test-utils';
import * as stories from './MediaTile.stories';

describe('MediaTile tests', () => {
  testAllStories(stories);
});
