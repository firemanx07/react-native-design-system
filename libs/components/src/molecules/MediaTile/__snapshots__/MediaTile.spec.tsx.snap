// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaTile tests MediaTile snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    MediaTile:
  </Text>
  <View
    style={
      [
        {
          "borderColor": "#222222",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      size={124}
      style={
        [
          {
            "height": 124,
            "overflow": "hidden",
            "width": 124,
          },
        ]
      }
    >
      <Image
        size={124}
        source={
          {
            "uri": "https://picsum.photos/124/124",
          }
        }
        style={
          [
            {
              "height": 124,
              "width": 124,
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    fontFamily="Muli-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "color": "#666666",
          "marginBottom": 4,
        },
      ]
    }
  >
    Sample variants:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        New - Comment
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <View
          borderRadius={4}
          size={18}
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FF8066",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 18,
                "justifyContent": "center",
                "minWidth": 18,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              {
                "backgroundColor": "#2FBFA7",
                "paddingHorizontal": 6,
                "position": "absolute",
                "right": 6,
                "top": 6,
                "zIndex": 1,
              },
            ]
          }
        >
          <Text
            color="#FFFFFF"
            fontFamily="Muli-ExtraBold"
            fontSize={10}
            lineHeight={14}
            style={
              [
                {
                  "color": "#222222",
                  "fontFamily": "Muli-ExtraBold",
                  "fontSize": 10,
                  "lineHeight": 14,
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "letterSpacing": 1.5,
                },
              ]
            }
          >
            NEW
          </Text>
        </View>
        <test-file-stub
          fill="#FFFFFF"
          height={24}
          style={
            [
              {
                "bottom": 4,
                "position": "absolute",
                "right": 4,
                "zIndex": 1,
              },
            ]
          }
          width={24}
        />
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        New - Selectable
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <View
          borderRadius={4}
          size={18}
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FF8066",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 18,
                "justifyContent": "center",
                "minWidth": 18,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              {
                "backgroundColor": "#2FBFA7",
                "paddingHorizontal": 6,
                "position": "absolute",
                "right": 6,
                "top": 6,
                "zIndex": 1,
              },
            ]
          }
        >
          <Text
            color="#FFFFFF"
            fontFamily="Muli-ExtraBold"
            fontSize={10}
            lineHeight={14}
            style={
              [
                {
                  "color": "#222222",
                  "fontFamily": "Muli-ExtraBold",
                  "fontSize": 10,
                  "lineHeight": 14,
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "letterSpacing": 1.5,
                },
              ]
            }
          >
            NEW
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(34,34,34,0.8)",
              "borderColor": "#FFFFFF",
              "borderRadius": 10,
              "bottom": 6,
              "height": 20,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 6,
              "width": 20,
              "zIndex": 3,
            }
          }
        />
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        New - Selected
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <View
          borderRadius={4}
          size={18}
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FF8066",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 18,
                "justifyContent": "center",
                "minWidth": 18,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              {
                "backgroundColor": "#2FBFA7",
                "paddingHorizontal": 6,
                "position": "absolute",
                "right": 6,
                "top": 6,
                "zIndex": 1,
              },
            ]
          }
        >
          <Text
            color="#FFFFFF"
            fontFamily="Muli-ExtraBold"
            fontSize={10}
            lineHeight={14}
            style={
              [
                {
                  "color": "#222222",
                  "fontFamily": "Muli-ExtraBold",
                  "fontSize": 10,
                  "lineHeight": 14,
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "letterSpacing": 1.5,
                },
              ]
            }
          >
            NEW
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "rgba(34,34,34,0.5)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 2,
              },
            ]
          }
        />
        <View
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FF8066",
              "borderColor": "#FF8066",
              "borderRadius": 10,
              "bottom": 6,
              "height": 20,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 6,
              "width": 20,
              "zIndex": 3,
            }
          }
        >
          <test-file-stub
            fill="#FFFFFF"
            height={16}
            style={
              [
                {},
              ]
            }
            width={16}
          />
        </View>
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Video
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <Text
          fontFamily="Muli-Bold"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 12,
                "lineHeight": 18,
              },
              {
                "color": "#FFFFFF",
                "left": 8,
                "position": "absolute",
                "textShadowColor": "rgba(34,34,34,0.5)",
                "textShadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "textShadowRadius": 2,
                "top": 5,
                "zIndex": 2,
              },
            ]
          }
        >
          01:22
        </Text>
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Video - New - Comment
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <View
          borderRadius={4}
          size={18}
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#FF8066",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 18,
                "justifyContent": "center",
                "minWidth": 18,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              {
                "backgroundColor": "#2FBFA7",
                "paddingHorizontal": 6,
                "position": "absolute",
                "right": 6,
                "top": 6,
                "zIndex": 1,
              },
            ]
          }
        >
          <Text
            color="#FFFFFF"
            fontFamily="Muli-ExtraBold"
            fontSize={10}
            lineHeight={14}
            style={
              [
                {
                  "color": "#222222",
                  "fontFamily": "Muli-ExtraBold",
                  "fontSize": 10,
                  "lineHeight": 14,
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "letterSpacing": 1.5,
                },
              ]
            }
          >
            NEW
          </Text>
        </View>
        <test-file-stub
          fill="#FFFFFF"
          height={24}
          style={
            [
              {
                "bottom": 4,
                "position": "absolute",
                "right": 4,
                "zIndex": 1,
              },
            ]
          }
          width={24}
        />
        <Text
          fontFamily="Muli-Bold"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 12,
                "lineHeight": 18,
              },
              {
                "color": "#FFFFFF",
                "left": 8,
                "position": "absolute",
                "textShadowColor": "rgba(34,34,34,0.5)",
                "textShadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "textShadowRadius": 2,
                "top": 5,
                "zIndex": 2,
              },
            ]
          }
        >
          01:22
        </Text>
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Unlock button
      </Text>
      <View
        size={124}
        style={
          [
            {
              "height": 124,
              "overflow": "hidden",
              "width": 124,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(34,34,34,0.5)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 2,
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 3,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "#FFFFFF",
                "borderRadius": 16,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 28,
                "justifyContent": "center",
                "maxWidth": "100%",
                "opacity": 1,
                "paddingHorizontal": 8,
              }
            }
          >
            <test-file-stub
              fill="#FFFFFF"
              height={16}
              style={
                [
                  {},
                ]
              }
              width={16}
            />
            <Text
              color="rgba(34,34,34,0.3)"
              fontFamily="Muli-Bold"
              fontSize={14}
              hasIcon={true}
              lineHeight={20}
              numberOfLines={1}
              size="Small"
              style={
                [
                  {
                    "color": "#222222",
                    "fontFamily": "Muli-Bold",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "color": "rgba(34,34,34,0.3)",
                    "marginLeft": 4,
                  },
                  {
                    "color": "#FFFFFF",
                  },
                ]
              }
            >
              Freishalten
            </Text>
          </View>
        </View>
        <Image
          size={124}
          source={
            {
              "uri": "https://picsum.photos/124/124",
            }
          }
          style={
            [
              {
                "height": 124,
                "width": 124,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
